aten/src/ATen/native/Linear.cpp

```diff
@@ -541,9 +541,17 @@ Tensor einsum(c10::string_view equation, TensorList operands, at::OptionalIntArr
 
   // Sum out contraction dims
   if (perm_index - out_num_dim > 0) {
-    std::vector<int64_t> sum_dims(perm_index - out_num_dim);
-    std::iota(sum_dims.begin(), sum_dims.end(), out_num_dim);
-    ops[0] = ops[0].sum(sum_dims);
+    if (num_ops > 1) {
```

Comment:
This if is for the special case where there is a single input and thus all the code above was a noop and so we need to actually do the reduction here.
For all the other cases (>1), we are guaranteed from the code above that all reduced dimensions are now of size 1 and thus can just be viewed the way we want?

Reply:
Precisely

Reply:
A comment would be great for future readers.

Reply:
oh shoot i just saw this, will add in another pr